contracts/interchain-token-service/src/contract.rs

```diff
@@ -140,53 +142,51 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
         token_meta_data: TokenMetadata,
         initial_supply: i128,
         minter: Option<Address>,
-    ) -> Result<Address, ContractError> {
+    ) -> Result<(Address, BytesN<32>), ContractError> {
         caller.require_auth();
 
-        let mut initial_minter: Option<Address> = None;
-
-        if initial_supply > 0 {
-            initial_minter = Some(env.current_contract_address());
-        } else if let Some(ref minter_addr) = minter {
+        let initial_minter = if initial_supply > 0 {
+            Some(env.current_contract_address())
+        } else if let Some(ref minter) = minter {
             ensure!(
-                *minter_addr != env.current_contract_address(),
+                *minter != env.current_contract_address(),
                 ContractError::InvalidMinter
             );
-            initial_minter = minter.clone();
-        }
+            Some(minter.clone())
+        } else {
+            None
+        };
 
-        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt.clone());
-        let token_id = Self::interchain_token_id(env, caller.clone(), salt);
+        let deploy_salt = Self::interchain_token_deploy_salt(env, caller.clone(), salt);
+        let token_id = Self::interchain_token_id(env, None, deploy_salt);
 
         let deployed_address = env
             .deployer()
-            .with_address(caller.clone(), deploy_salt)
+            .with_address(env.current_contract_address(), token_id.clone())
             .deploy_v2(
                 Self::interchain_token_wasm_hash(env),
                 (
                     env.current_contract_address(),
                     initial_minter,
                     env.current_contract_address(),
-                    token_id,
+                    token_id.clone(),
                     token_meta_data,
                 ),
             );
 
         if initial_supply > 0 {
             let token = InterchainTokenClient::new(env, &deployed_address);
 
-            // TODO: tokenManager, registeredTokenAddress, deployedTokenManager, removeFlowLimiter, addFlowLimiter
+            // AXE-6858: the tokenManager related logic needs to be implemented here.
```

Comment:
```suggestion
```
i think it's fine to the link to the code from the task since we're tracking in JIRA and not the code itself